contracts/Controller.sol

```diff
@@ -418,7 +420,34 @@ contract Controller is ReentrancyGuard, Ownable {
      * @dev only account owner or operator can withdraw long option from vault
      * @param _args WithdrawArgs structure
      */
-    // function _withdrawCollateral(Actions.WithdrawArgs memory _args) internal isAuthorized(_args.owner) {}
+    function _withdrawCollateral(Actions.WithdrawArgs memory _args)
+        internal
+        isAuthorized(msg.sender, _args.owner)
+        returns (MarginAccount.Vault memory)
+    {
+        require(checkVaultId(_args.owner, _args.vaultId), "Controller: invalid vault id");
+
+        MarginAccount.Vault memory vault = vaults[_args.owner][_args.vaultId];
+        if ((vault.shortOtokens.length > 0) && (vault.shortOtokens[0] != address(0))) {
```

Comment:
better have an internal function for this logic, maybe called `_notEmpty`
`arr.length > 0 && arr[0] !=address(0) `